integration-tests/elasticsearch-rest-client/src/main/java/io/quarkus/it/elasticsearch/FruitService.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.it.elasticsearch;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+@ApplicationScoped
+public class FruitService {
+    @Inject
+    RestClient restClient;
+
+    public void index(Fruit fruit) throws IOException {
+        Request request = new Request(
+                "PUT",
+                "/fruits/_doc/" + fruit.id);
+        request.setJsonEntity(JsonObject.mapFrom(fruit).toString());
+        restClient.performRequest(request);
+    }
+
+    public Fruit get(String id) throws IOException {
+        Request request = new Request(
+                "GET",
+                "/fruits/_doc/" + id);
+        Response response = restClient.performRequest(request);
+        String responseBody = EntityUtils.toString(response.getEntity());
+        JsonObject json = new JsonObject(responseBody);
+        return json.getJsonObject("_source").mapTo(Fruit.class);
+    }
+
+    public List<Fruit> searchByColor(String color) throws IOException {
+        return search("color", color);
+    }
+
+    public List<Fruit> searchByName(String name) throws IOException {
+        return search("name", name);
+    }
+
+    private List<Fruit> search(String term, String match) throws IOException {
+        Request request = new Request(
+                "GET",
+                "/fruits/_search");
+        String query = String.format("{\"query\": { \"match\": { \"%s\":\"%s\"}}}", term, match);
```

Comment:
I don't know if you intend to use the same code for the quickstart, but in case you do... this code is prone to injection. Users could potentially insert some very heavy aggregation in your query, for example. I don't know for sure, but I can imagine building on this vulnerability to trigger a DOS. 

The proper way to do this would be to use a library such as GSON, Jackson, or whatever you fancy to build a structured JSON object, then serializing that object to a string. But then the code gets ugly, there's no denying that :)

Reply:
Agreed, that can't happen in the quickstart as people might use it. As you already use Jackson and Vert.x JsonObject, looks like you could use that.